### PR TITLE
add payment fields component spec

### PR DESCRIPTION
**Pull request by andyfen**
spec for paymentfields component

### CODE REVIEW

File: spec/client/javascript/components/payment-fields/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Payment Fields
```
Comment (crookedneighbor) [inline spec/client/javascript/components/payment-fields/README.md:1]: Can we add something that gives an explanation for what these are?

```suggestion
# Payment Fields

This component allows customers to pay with payment methods that are only available in certain locations.
```

File: spec/client/javascript/components/payment-fields/styling.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  active: {
+    color: "black",
+    backgroundColor: "white",
+  },
+  invalid: {
+    color: "#d20000",
+    backgroundColor: "white",
+  }
+};
+
```
Comment (crookedneighbor) [inline spec/client/javascript/components/payment-fields/styling.md:49]:
```suggestion
```

**Review by crookedneighbor — CHANGES_REQUESTED** [review]

File: spec/client/javascript/components/payment-fields/styling.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# Style PaymentFields
+
+## Supported Styles
+
+### Properties
```
Comment (bluepnume) [inline spec/client/javascript/components/payment-fields/styling.md:5]: Can you work with @leogedler to agree on a common naming format and set of properties between these fields and the card fields? Would be good to make sure they're consistent: https://github.com/paypal/paypal-sdk-spec/blob/main/spec/client/javascript/components/card-fields/styling.md

**Comment by bluepnume** [comment]
This is looking great, thanks @andyfen!

In some of the examples can you show how this works end-to-end; e.g. with the button component, rendering, getting a click, creating an order and getting a successful approval?

File: spec/client/javascript/components/payment-fields/styling.md (hunk 3 above)
Comment (leogedler) [reply spec/client/javascript/components/payment-fields/styling.md:5]: @andyfen We have here the most updated styles supported for card-fields: https://github.com/paypal/paypal-smart-payment-buttons/blob/master/src/card/constants.js#L29

And here an example on how we are passing the style object into the cardFields component: https://github.com/paypal/paypal-sdk-samples/pull/1/files#diff-0a7b080f4b388288a16e65bf73e045b1f6b6402151baab2a13485aa33780ac1eR135

File: spec/client/javascript/components/payment-fields/README.md (hunk 1 above)
Comment (crookedneighbor) [reply spec/client/javascript/components/payment-fields/README.md:1]: Love the explanation you added. 

Is there some context on why the name "Payment Fields" was chosen for this product? It sounds pretty generic, and I'd never guess that these localized payment methods would be included under it. (not that I have a good name to replace it with, just curious)

**Pull request closed without merging**